compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -523,14 +523,14 @@ extern "C" typedef void (*LLVMRustSelfProfileBeforePassCallback)(void*, // LlvmS
 extern "C" typedef void (*LLVMRustSelfProfileAfterPassCallback)(void*); // LlvmSelfProfiler
 
 std::string LLVMRustwrappedIrGetName(const llvm::Any &WrappedIr) {
-  if (any_isa<const Module *>(WrappedIr))
-    return any_cast<const Module *>(WrappedIr)->getName().str();
-  if (any_isa<const Function *>(WrappedIr))
-    return any_cast<const Function *>(WrappedIr)->getName().str();
-  if (any_isa<const Loop *>(WrappedIr))
-    return any_cast<const Loop *>(WrappedIr)->getName().str();
-  if (any_isa<const LazyCallGraph::SCC *>(WrappedIr))
-    return any_cast<const LazyCallGraph::SCC *>(WrappedIr)->getName();
+  if (const auto *Cast = any_cast<const Module *>(WrappedIr))
```

Comment:
I'm pretty sure this is the wrong `any_cast` overload -- the checked one needs a pointer argument like `&WrappedIR`. We're roughly mimicking this function:
https://github.com/rust-lang/llvm-project/blob/ea6fa9c2d43aaf0f11559719eda9b54d356d5416/llvm/lib/Passes/StandardInstrumentations.cpp#L189

... which changed from `any_isa` to `any_cast` here:
https://github.com/rust-lang/llvm-project/commit/bb7940e25f6ca201ca57943544016390f1d2e504#diff-9bcb7cbad2f3a6a1516397a3d059d03c91dcfdbd9edda564baddf1bb2652d928R189

Reply:
Ah, thanks, sorry for making such simple mistake. Codegen/codegen-units tests did not catch it.